src/openms/source/KERNEL/MSExperiment.cpp

```diff
@@ -214,91 +217,86 @@ namespace OpenMS
   */
-  void MSExperiment::updateRanges(Int ms_level)
+  void MSExperiment::updateRanges()
   {
     #ifdef OPENMS_ASSERTIONS
-      double rt_min = RangeRT::isEmpty() ? 0 : getMinRT();
-      double rt_max = RangeRT::isEmpty() ? 0 : getMaxRT();
-      double mz_min = RangeMZ::isEmpty() ? 0 : getMinMZ();
-      double mz_max = RangeMZ::isEmpty() ? 0 : getMaxMZ();
-      double int_min = RangeIntensity::isEmpty() ? 0 : getMinIntensity();
-      double int_max = RangeIntensity::isEmpty() ? 0 : getMaxIntensity();
-      double im_min = RangeMobility::isEmpty() ? 0 : getMinMobility();
-      double im_max = RangeMobility::isEmpty() ? 0 : getMaxMobility();
+      double rt_min = combined_ranges_.RangeRT::isEmpty() ? 0 : combined_ranges_.getMinRT();
+      double rt_max = combined_ranges_.RangeRT::isEmpty() ? 0 : combined_ranges_.getMaxRT();
+      double mz_min = combined_ranges_.RangeMZ::isEmpty() ? 0 : combined_ranges_.getMinMZ();
+      double mz_max = combined_ranges_.RangeMZ::isEmpty() ? 0 : combined_ranges_.getMaxMZ();
+      double int_min = combined_ranges_.RangeIntensity::isEmpty() ? 0 : combined_ranges_.getMinIntensity();
+      double int_max = combined_ranges_.RangeIntensity::isEmpty() ? 0 : combined_ranges_.getMaxIntensity();
+      double im_min = combined_ranges_.RangeMobility::isEmpty() ? 0 : combined_ranges_.getMinMobility();
+      double im_max = combined_ranges_.RangeMobility::isEmpty() ? 0 : combined_ranges_.getMaxMobility();
     #endif
 
-    // reset mz/rt/int range
-    this->clearRanges();
+    // Reset all range managers
+    clearRanges();
 
-    // empty
+    // Empty experiment
     if (spectra_.empty() && chromatograms_.empty())
     {
       return;
     }
 
-    // update
+    // Update spectrum ranges
     for (Base::iterator it = spectra_.begin(); it != spectra_.end(); ++it)
-    {
-      if (ms_level < Int(0) || Int(it->getMSLevel()) == ms_level)
-      {
-        // ranges
-        this->extendRT(it->getRT()); // RT
-        // m/z, intensity and ion mobility from spectrum's range
-        it->updateRanges();
-        this->extend(*it);
-      }
-      // for MS level = 1 we extend the range for all the MS2 precursors
-      if (ms_level == 1 && it->getMSLevel() == 2)
+    {      
+      // Update ranges for the spectrum itself
+      it->updateRanges();
+      
+      // Update spectrum range manager with this spectrum's ranges
+      // Add to both general ranges and MS level-specific ranges
+      spectrum_ranges_.extendUnsafe(*it, 0);
+      spectrum_ranges_.extendRT(it->getRT(), 0); // RT is not part of the range of an individual spectrum
+      
+      spectrum_ranges_.extendUnsafe(*it, it->getMSLevel());
+      spectrum_ranges_.extendRT(it->getRT(), it->getMSLevel()); // RT is not part of the range of an individual spectrum
+      
+      // For MS1 level, also include MS2 precursor m/z values
+      if (it->getMSLevel() == 2 && !it->getPrecursors().empty())
```

Comment:
including precursor mz was done before so they are considered for the overall spectra range